docs/linux_dependencies.md

```diff
@@ -45,48 +45,44 @@ sudo xbps-install -S pkgconf alsa-lib-devel libX11-devel eudev-libudev-devel
 
 ## NixOS
 
-Add a `build.rs` file to your project with the following:
+Add a `build.rs` file to your project:
 
 ```rust
+$ cat > build.rs <<EOF
 fn main() {
     if cfg!(target_os = "linux") {
         println!("cargo:rustc-link-lib=vulkan");
     }
 }
+EOF
 ```
 
 These packages provide the dependencies required to run a bevy project. They can be installed globally or via nix-shell.
 
-`nix-shell -p pkgconfig x11 xorg.libXcursor xorg.libXrandr xorg.libXi vulkan-tools lutris vulkan-headers vulkan-loader vulkan-validation-layers alsaLib`
+`nix-shell -p cargo pkgconfig udev alsaLib x11 xorg.libXcursor xorg.libXrandr xorg.libXi vulkan-tools lutris vulkan-headers vulkan-loader vulkan-validation-layers`
 
-Alternatively, you can copy the following code block and create a file called `shell.nix`. You can now enter nix-shell just by running `nix-shell`.
+Alternatively, you can define `shell.nix` and enther it by running `nix-shell`.
 
 ```nix
-# shell.nix
+$ cat > shell.nix <<EOF
 
 { pkgs ? import <nixpkgs> { } }:
-
-pkgs.mkShell {
+with pkgs;
+mkShell {
   buildInputs = [
-    pkgs.alsaLib
-    pkgs.lutris
-    pkgs.pkgconfig
-    pkgs.vulkan-headers
-    pkgs.vulkan-loader
-    pkgs.vulkan-tools
-    pkgs.vulkan-validation-layers
-    pkgs.x11
-    pkgs.xorg.libXcursor
-    pkgs.xorg.libXi
-    pkgs.xorg.libXrandr
+    cargo
+    pkgconfig udev alsaLib
+    x11 xorg.libXcursor xorg.libXrandr xorg.libXi
+    vulkan-tools lutris vulkan-headers vulkan-loader vulkan-validation-layers
   ];
 }
```

Comment:
here `cargo` and `udev` were added. NixOS is all about having explicit dependencies so assuming there is no global installation of `cargo` and `udev` seems to be more natural. I also shrieked related stuff into single lines so we don't take so excessive amount of lines within this document.